evalml/data_checks/invalid_targets_data_check.py

```diff
@@ -59,25 +59,24 @@ def validate(self, X, y):
             raise ValueError("y cannot be None")
 
         y = _convert_to_woodwork_structure(y)
-        y = _convert_woodwork_types_wrapper(y.to_series())
-
-        null_rows = y.isnull()
+        valid_target_types = numeric_and_boolean_ww + categorical_ww_types
+        if y.logical_type not in numeric_and_boolean_ww + categorical_ww_types:
+            messages["errors"].append(DataCheckError(message="Target is unsupported {} Woodwork type. Valid Woodwork logical types include: {}"
```

Comment:
I'm conflicted about whether or not we should mention woodwork here. On the one hand, we use woodwork under the hood, so people are going to have to learn about it if they want to work with how evalml handles typing. On the other hand, maybe this error message is easier to understand when it doesn't mention woodwork, since some users may not realize woodwork is in the mix until they see this. I guess the question is, what message makes it easiest for users to understand the problem and make a fix?

I think a good compromise is to say `"Target is unsupported {} type. Valid Woodwork logical types include: {}"`